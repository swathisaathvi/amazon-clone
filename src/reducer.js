import { act } from "react-dom/test-utils";

export const initialState={
    basket: [
        // {
        //     id: "238724",
        //     title: "New Apple ipad pro (12.9) inch",
        //     price: 599.98,
        //     rating: 4,
        //     image: "https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._SX325_BO1,204,203,200_.jpg"
        // },
        // {
        //     id: "325325",
        //     title: "Watch new apple series generation 5",
        //     price: 239.98,
        //     rating: 3,
        //     image: "https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._SX325_BO1,204,203,200_.jpg"
        // }
],
    user: null
}

export const getBasketTotal = (basket) => {
    return basket?.reduce((amount, item) => item.price + amount, 0);
}

const reducer = (state, action) => {
    switch(action.type) {
        case 'ADD_TO_BASKET': 
        //Logic for adding item to basket
            return {
                ...state,
                basket: [...state.basket, action.item]
            }
            break;
        case 'REMOVE_FROM_BASKET':
        //Logic for removing item from basket
            let newBasket = [...state.basket];
            const index = newBasket.findIndex((basketItem)=> basketItem.id === action.id)
            if (index >=0){
                newBasket.splice(index, 1);
            }
            return {...state, 
                basket: newBasket}
            break;
        default:
            return state; 
    }
}

export default reducer;