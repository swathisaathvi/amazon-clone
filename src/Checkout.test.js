import React from "react";
import { fireEvent, render, screen } from '@testing-library/react';
import Checkout from "./Checkout";
import { StateProvider } from "./StateProvider";
import reducer from "./reducer";

const MockedStateProvider = ({ children }) => (
  <StateProvider initialState={{ basket: [
    { id: 1, title: "Item 1", image: "item1.jpg", price: 10, rating: 4 },
    { id: 2, title: "Item 2", image: "item2.jpg", price: 15, rating: 5 },
  ] }} reducer={reducer}>
    {children}
  </StateProvider>
);

test("performs Checkout page snapshot testing", () => {
  const { asFragment } = render(
    <MockedStateProvider>
      <Checkout />
    </MockedStateProvider>
  );

  expect(screen.getByText("Item 1")).toBeInTheDocument;
  expect(screen.getByText("Item 2")).toBeInTheDocument;
  
  //const remFromBsktBtn = screen.getAllByText("Remove From Basket");
  //fireEvent.click(remFromBsktBtn[1]);

  expect(asFragment()).toMatchSnapshot();
});
