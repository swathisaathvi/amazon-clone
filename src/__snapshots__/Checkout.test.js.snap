// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs Checkout page snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="checkout"
  >
    <div
      class="checkout__left"
    >
      <img
        alt=""
        class="checkout__ad"
        src="https://images-na.ssl-images-amazon.com/images/G/02/UK_CCMP/TM/OCC_Amazon1._CB423492668_.jpg"
      />
      <div
        class="checkout__title"
      >
        <h2>
          Your Shopping Basket
        </h2>
        <div
          class="checkoutProduct"
        >
          <img
            alt=""
            class="checkoutProduct__image"
            src="item1.jpg"
          />
          <div
            class="checkoutProduct__info"
          >
            <p
              class="checkoutProduct__title"
            >
              Item 1
            </p>
            <p
              class="checkoutProduct__price"
            >
              <small>
                $
              </small>
              <strong>
                10
              </strong>
            </p>
            <div
              class="checkoutProduct__rating"
            >
              <p>
                ⭐
              </p>
              <p>
                ⭐
              </p>
              <p>
                ⭐
              </p>
              <p>
                ⭐
              </p>
            </div>
            <button>
              Remove From Basket
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="checkout__right"
    >
      <div
        class="subtotal"
      >
        <p
          data-testid="subtotal"
        >
          Subtotal (1 items) : 
          <strong>
            $10
          </strong>
        </p>
        <small
          class="subtotal__gift"
        >
          <input
            type="checkbox"
          />
           This order contains the gift service
        </small>
        <button>
          Proceed to Checkout
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
